repo_resource/common.py

```diff
@@ -76,8 +129,10 @@ def source_config_from_payload(payload):
     p = SourceConfiguration(**payload['source'])
     source_url = urlparse(p.url)
 
-    if source_url.netloc == 'gitlab.com' and \
-       (source_url.scheme == 'http' or source_url.scheme == 'https'):
+    if (
+        source_url.netloc == 'gitlab.com' and
+        re.fullmatch('https?', source_url.scheme)
```

Comment:
Seems like a good improvement, but it's not relevant to this patch.

This patch is "use git ls-remote for fetching revision"

This diff is "improve url matching regex".

Please split this out to a seperate commit.

Reply:
Done